pkg/controllers/auditlog/auditpolicy/controller.go

```diff
@@ -0,0 +1,75 @@
+package auditpolicy
+
+import (
+	"context"
+	"fmt"
+
+	auditlogv1 "github.com/rancher/rancher/pkg/apis/auditlog.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/auth/audit"
+	"github.com/rancher/rancher/pkg/generated/controllers/auditlog.cattle.io"
+	v1 "github.com/rancher/rancher/pkg/generated/controllers/auditlog.cattle.io/v1"
+)
+
+type handler struct {
+	auditpolicy v1.AuditPolicyController
+	writer      *audit.Writer
+}
+
+func (h *handler) OnChange(key string, obj *auditlogv1.AuditPolicy) (*auditlogv1.AuditPolicy, error) {
+	if !obj.Spec.Enabled {
+		h.writer.RemovePolicy(obj)
+
+		obj.Status = auditlogv1.AuditPolicyStatus{
+			Condition: auditlogv1.AuditPolicyStatusConditionInactive,
+		}
+
+		if _, err := h.auditpolicy.UpdateStatus(obj); err != nil {
+			return obj, fmt.Errorf("could not mark audit log policy '%s/%s' as disabled: %s", obj.Namespace, obj.Name, err)
+		}
+
+		return obj, nil
+	}
+
+	if err := h.writer.UpdatePolicy(obj); err != nil {
+		obj.Status = auditlogv1.AuditPolicyStatus{
+			Condition: auditlogv1.AuditPolicyStatusConditionInvalid,
+			Message:   err.Error(),
+		}
+
+		if _, err := h.auditpolicy.UpdateStatus(obj); err != nil {
```

Comment:
We should not ignore the returned object from `UpdateStatus` because the update call changes the object (it changes the `.status` field but also changes the resource version). We want to give that object to the next handlers that will run so that it has the "latest" version of the auditpolicy, not a stale one. (This is more for future proofing, I'm aware there are no other handlers running)